src/ws/session.c

```diff
@@ -525,13 +525,23 @@ perform_gssapi (const char *rhost,
 static int
 session (char **env)
 {
-  char *argv[] = { "cockpit-bridge", NULL };
+  char *argv[] = { NULL /* user's shell */, "-c", "exec cockpit-bridge", NULL };
   gss_key_value_set_desc store;
   struct gss_key_value_element_struct element;
   OM_uint32 major, minor;
   krb5_context k5;
+  struct passwd *pwd;
   int res;
 
+  /* determine the user's shell and run the bridge through it, so that we catch
+   * disabled accounts like /bin/false or /bin/nologin shells */
+  pwd = getpwuid (geteuid ());
+  if (!pwd)
+    errx (EX, "failed to get passwd entry for user: %m");
+  argv[0] = pwd->pw_shell;
+  if (!argv[0])
+    errx (EX, "passwd entry for user has NULL shell");
```

Comment:
Shouldn't this be the same case as having a disabled shell? ie: return 1 not EX?

Reply:
There is no concept of "disabled shell" in /etc/passwd, AFAICS. You can leave the shell field empty or not even specify the last field at all, and in both cases `pw_shell == ""`. So `NULL` is really an exceptional Should Not Happen™ case, I have no idea how it would get triggered. So I think `EX` is quite fine.